File: sources/java/_08_average.java
// task 08 average — expected output: 0.498046875
// build: javac _08_average.java    run: java _08_average
// build (graalvm native-image): native-image -O2 _08_average    run: ./_08_average
// Double.toString is locale independent and prints the shortest round-tripping decimal.

public class _08_average {
    public static void main(String[] args) {
        double total = 0.0;
        for (long i = 0; i < 100000000L; i++) {
            double reading = (i % 256) / 256.0;
            total += reading;
        }
        System.out.println(Double.toString(total / 100000000));
    }
}

File: sources/python/08_average.py
# task 08 average — expected output: 0.498046875
# build: python 08_average.py | pypy 08_average.py | graalpy 08_average.py | nuitka --standalone 08_average.py    run: python 08_average.py
# note: every reading is a multiple of 1/256 and the total stays under 2^53, so the sum is exact.

total = 0.0

for i in range(100000000):
    reading = (i % 256) / 256.0
    total += reading

print(total / 100000000)
